source/library/Witch/Instances.hs

```diff
@@ -416,27 +378,17 @@ instance TryFrom.TryFrom Integer Natural.Natural where
   tryFrom = Utility.eitherTryFrom $
     \s -> if s < 0 then Left Exception.Underflow else Right $ fromInteger s
 
--- | Uses 'fromIntegral' when the input is between -16,777,215 and 16,777,215
--- inclusive.
+-- | TODO
 instance TryFrom.TryFrom Integer Float where
-  tryFrom = Utility.eitherTryFrom $ \s ->
-    if s < -maxFloat
-      then Left Exception.Underflow
-      else
-        if s > maxFloat
-          then Left Exception.Overflow
-          else Right $ fromIntegral s
-
--- | Uses 'fromIntegral' when the input is between -9,007,199,254,740,991 and
--- 9,007,199,254,740,991 inclusive.
+  tryFrom = Utility.eitherTryFrom $ \s -> case fromInteger s of
+    t | s == truncate t -> Right t
+    _ -> Left Exception.LossOfPrecision
```

Comment:
This is the change in behavior. I can't use any `FiniteBits` tricks because `Integer` is arbitrary precision.